zio/zsonio/writer.go

```diff
@@ -54,362 +28,13 @@ func (w *Writer) Close() error {
 }
 
 func (w *Writer) Write(rec *zng.Record) error {
-	typ, err := w.mapper.Translate(rec.Type)
+	s, err := w.formatter.FormatRecord(rec)
 	if err != nil {
 		return err
 	}
-	if err := w.writeValueAndDecorate(typ, rec.Raw); err != nil {
-		return err
-	}
-	return w.write("\n")
-}
-
-func (w *Writer) writeValueAndDecorate(typ zng.Type, bytes zcode.Bytes) error {
-	known := w.typedefs.exists(typ)
-	if err := w.writeValue(0, typ, bytes, known, false); err != nil {
+	if _, err := w.writer.Write([]byte(s)); err != nil {
```

Comment:
Avoid copying s if w.writer implements io.StringWriter.
```suggestion
	if _, err := io.WriteString(w.writer.Write, s); err != nil {
```